File: JavaScript/Queues/Queue.js
class Node {
    constructor(val){
        this.val = val;
        this.next = null;
    }
}

class Queue {
    constructor(){
        this.first = null;
        this.last = null;
        this.size = 0;
    }

    enqueue(val){
        let node = new Node(val);
        if (this.size === 0){
            this.first = node;
            this.last = node;
        }
        else {
            this.last.next = node;
            this.last = node;
        }
        this.size += 1;
        return this.size;
        
    }

    dequeue(){
        if(this.size === 0){
            return null;
        } 
        let temp = this.first;
        if (this.size === 1){
            this.first = null;
            this.last = null;
        }
        else {
            let newHead = temp.next;
            temp.next = null;
            this.first = newHead;
        }
        this.size -= 1;
        return temp.val;
    }
}

let q = new Queue()

console.log(q.enqueue("FIRST"))
console.log(q)
console.log(q.enqueue("SECOND"))
console.log(q)
console.log(q.enqueue("THIRD"))
console.log(q)
console.log(q.dequeue())
console.log(q)
console.log(q.dequeue())
console.log(q)
console.log(q.dequeue())
console.log(q)

File: Python/Queues/Queue.py
class Node:
    def __init__(self,val):
        self.val = val
        self.next = None

class Queue:
    def __init__(self):
        self.first = None
        self.last = None
        self.size = 0

    def enqueue(self, val):
        newNode = Node(val)
        if self.size == 0:
            self.first = newNode
            self.last = newNode
        else:
            self.last.next = newNode
            self.last = newNode
        self.size += 1
        return self

    def dequeue(self):
        if self.size == 0:
            return None
        currentHead = self.first
        if self.size == 1:
            self.first = None
            self.last = None
        else:
            newHead = currentHead.next
            currentHead.next = None
            self.first = newHead
        self.size -= 1
        return currentHead.val

    def getNode(self,idx):
        if idx < 0 or idx >= self.size:
            return None
        current = self.first
        cnt = 1
        while cnt <= idx:
            current = current.next
            cnt += 1
        return current.val

    def getSize(self):
        return self.size

queue = Queue()
queue.enqueue(5)
queue.enqueue(6)
queue.enqueue(7)
print(queue.getSize())
print(queue.getNode(0))
print(queue.getNode(1))
print(queue.getNode(2))
print(queue.getNode(3))
print(queue.dequeue())
print(queue.dequeue())
print(queue.dequeue())
print(queue.dequeue())